resource_transip_vps.go

```diff
@@ -296,6 +300,28 @@ func resourceVpsRead(d *schema.ResourceData, m interface{}) error {
 	return nil
 }
 
+func resourceVpsUpdate(d *schema.ResourceData, m interface{}) error {
+	client := m.(repository.Client)
+	repository := vps.Repository{Client: client}
+
+	vps, err := repository.GetByName(d.Id())
+	if err != nil {
+		return fmt.Errorf("failed to get details for vps %s with id %q: %s", d.Get("description").(string), d.Id(), err)
+	}
+
+	vps.Description = d.Get("description").(string)
+	vps.IsCustomerLocked = d.Get("is_customer_locked").(bool)
+	// Below results in a conversion error: interface conversion: interface {} is []interface {}, not []string
+	// vps.Tags = d.Get("tags").([]string)
+
+	err = repository.Update(vps)
```

Comment:
Can we make the update itself conditional on if one of the updateable attributes has changed? This will prevent a API call to Transip when there is no need for one.

Reply:
Checked if either description or lock updated